ui/basic-auth.js

```diff
@@ -0,0 +1,18 @@
+/* Exports a single function, which installs a basic auth check if that's how
+ * we're configured. Uses a constant-time buffer comparison */
+import bufferEq from 'buffer-equal-constant-time';
+import httpAuth from 'http-auth';
+
+export default function (envCreds, expressApp) {
```

Comment:
Eep, sorry to miss this @theresaanna -- the library would prefer if we had credentials in a file (which doesn't work so well with twelve-factor-style environment variables), but they mention this style in [their examples](https://github.com/http-auth/http-auth#custom-authentication).

This uses constant-time string comparison for the reasons I mentioned in http-auth/http-auth#78